hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/MessageConversionExceptionHandler.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2021 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.amqp;
+
+import org.eclipse.hawkbit.exception.ConditionalErrorHandler;
+import org.eclipse.hawkbit.exception.ErrorHandlerChain;
+import org.springframework.amqp.AmqpRejectAndDontRequeueException;
+import org.springframework.amqp.support.converter.MessageConversionException;
+
+/**
+ * An error handler for message conversion exception resulting from AMQP.
+ */
+public class MessageConversionExceptionHandler implements ConditionalErrorHandler<Throwable> {
+
+    @Override
+    public void doHandle(final Throwable t, final ErrorHandlerChain<Throwable> chain) {
+        if (t.getCause() instanceof MessageConversionException) {
+            throw new AmqpRejectAndDontRequeueException("The message could not be parsed", t.getCause());
```

Comment:
could you please verify if you could gather the relevant information from MessageConversionException by appending the messages from the throwable chain?

Reply:
Updated the changes here by extracting the relevant message from the cause